src/lib/one-graph/client.js

```diff
@@ -0,0 +1,835 @@
+    if (!enabledServicesInfo) {
+      console.warn('Unable to fetch enabled services for site for code generation')
+      return
+    }
+    const newEnabledServices = enabledServicesInfo.map((service) => service.service)
+    const enabledServicesCompareKey = enabledServices.sort().join(',')
+    const newEnabledServicesCompareKey = newEnabledServices.sort().join(',')
+
+    if (enabledServicesCompareKey !== newEnabledServicesCompareKey) {
+      log(
+        `${NETLIFYDEVLOG} ${chalk.magenta(
+          'Reloading',
+        )} Netlify Graph schema..., ${enabledServicesCompareKey} => ${newEnabledServicesCompareKey}`,
+      )
+      await refetchAndGenerateFromOneGraph({ netligraphConfig, state, netlifyToken, siteId })
+      log(`${NETLIFYDEVLOG} ${chalk.green('Reloaded')} Netlify Graph schema and regenerated functions`)
+    }
+  }
+
+  const close = () => {
+    shouldClose = true
+  }
+
+  let handle
+
+  const helper = async () => {
+    if (shouldClose) {
+      clearTimeout(handle)
+      onClose()
+    }
+
+    const next = await fetchCliSessionEvents({ appId, authToken, sessionId })
+
+    if (next.errors) {
+      next.errors.forEach((error) => {
+        onError(error)
+      })
+    }
+
+    const { events } = next
+
+    if (events.length !== 0) {
+      const ackIds = onEvents(events)
+      await ackCLISessionEvents({ appId, authToken, sessionId, eventIds: ackIds })
+    }
+
+    await enabledServiceWatcher(authToken, appId)
+
+    handle = setTimeout(helper, frequency)
+  }
+
+  // Fire immediately to start rather than waiting the initial `frequency`
+  helper()
+
+  return close
+}
+
+const createCLISession = async (netlifyToken, appId, name, metadata) => {
+  const payload = {
+    nfToken: netlifyToken,
+    appId,
+    name,
+    metadata,
+  }
+
+  const result = await fetchOneGraph(null, appId, internalOperationsDoc, 'CreateCLISessionMutation', payload)
+
+  const session =
+    result.data &&
+    result.data.oneGraph &&
+    result.data.oneGraph.createNetlifyCliSession &&
+    result.data.oneGraph.createNetlifyCliSession.session
+
+  return session
+}
+
+const updateCLISessionMetadata = async (netlifyToken, appId, sessionId, metadata) => {
+  const result = await fetchOneGraph(null, appId, internalOperationsDoc, 'UpdateCLISessionMetadataMutation', {
+    nfToken: netlifyToken,
+    sessionId,
+    metadata,
+  })
+
+  const session =
+    result.data &&
+    result.data.oneGraph &&
+    result.data.oneGraph.updateNetlifyCliSession &&
+    result.data.oneGraph.updateNetlifyCliSession.session
+
+  return session
+}
+
+const ackCLISessionEvents = async ({ appId, authToken, eventIds, sessionId }) => {
+  const result = await fetchOneGraph(null, appId, internalOperationsDoc, 'AckCLISessionEventMutation', {
+    nfToken: authToken,
+    sessionId,
+    eventIds,
+  })
+
+  const events = result.data && result.data.oneGraph && result.data.oneGraph.ackNetlifyCliEvents
```

Comment:
No need to assign to a var here and various other functions

Reply:
I prefer it since it gives a name and structure to the deeply-nested pluck.